src/main/java/dev/j3fftw/litexpansion/weapons/NanoBlade.java

```diff
@@ -29,12 +31,22 @@ public float getMaxItemCharge(ItemStack item) {
     @Override
     public ItemUseHandler getItemHandler() {
         return event -> {
+            ItemMeta NANOBLADEMETA;
             final Enchantment enchantment = Enchantment.getByKey(Constants.NANO_BLADE_ACTIVE_ENCHANT);
+            assert enchantment != null;
             final int removedLevel = event.getItem().removeEnchantment(enchantment);
 
             if (removedLevel == 0) {
                 event.getItem().addEnchantment(enchantment, 1);
+                NANOBLADEMETA = event.getItem().getItemMeta();
+                assert NANOBLADEMETA != null;
+                NANOBLADEMETA.setDisplayName(ChatColor.DARK_GREEN + "Nano Blade" + ChatColor.GREEN+ " (On)");
+            } else {
+                NANOBLADEMETA = event.getItem().getItemMeta();
+                assert NANOBLADEMETA != null;
```

Comment:
Remove these asserts and why are you setting the meta in these if else rather than just at the top?

Also, why is this full caps?? Naming conventions pls.

Reply:
that will copy meta before the enchant is added

Reply:
But that isn't needed?

Reply:
Not before the enchant is added. Then it will save the meta of the unenchanted nanoblade, enchant the nanoblade, rename the nanoblade, and then apply it. Nanoblade will be back in unenchanted state.

Reply:
I still don't see why you bother assinging here twice. If you're concerned about the enchant messing with the meta just use the ItemMeta to add/remove enchant instead of adding/removing from the item.

Reply:
kk ill do this